### ISSUE TITLE
Cannot get nilify to work with my forms

### ISSUE THREAD

**Issue by adrienbourgeois** [apotonick/reform#398]
I use the last version of reform (2.2.1) with rails

My form module:

```ruby
    module ContactModule
      include Reform::Form::Module
      property :contact, populate_if_empty: Contact do
        property :name, nilify: true
      end
    end
```

My contact form class:

```ruby
    class ContactForm < ::Reform::Form
      include ContactModule
    end
```

When I pass an empty for the name to that contact form, that empty string is saved in DB (and so not nilified)

Also when I try to define a type for my "name" property like so:

```ruby
    ...
        property :name, type: Types::Coercible::String, nilify: true
    ...
```

I've got this error:

```
`block in <module:ContactModule>': uninitialized constant ContactModule::Types (NameError)
```

Am I missing something?

**Comment by fran-worley** [comment]
You've got 2 issues here. 

1) nilify for which I will point you to @apotonick as that is part of the Disposable gem. I know that Nick is away at the moment

2) your types error. You need to make sure that a) you're including the coercible module and b) that you've actually installed the dry-types gem as reform won't do that for you.

You should check out the trailblazer gitter channel if you've got questions as the guys are always helping each other out and we are trying to keep this issue tracker for bug reports and fixes. I know that Nick is away at the moment so I'd try the channel see if anyone can help with your nilify question.

**Comment by adrienbourgeois** [comment]
Thanks for your help @fran-worley .

I finally got it working.
# First thing, I couldn't get the "nilify" option working without adding "dry-types" to my Gemfile.
# Then, indeed, I had to include de coercion module
# and finally I had to call "feature Coercion" in my form class. About that, I couldn't call that feature method in my form module. @apotonick don't you think that "feature" should be available when you include "Reform::Form::Module"?

So in summary, in your Gemfile add:

```ruby
gem 'dry-types', '~> 0.8.1'
```

You also have to explicitly require the coercion module (I did it in a initialiser):

```ruby
require 'disposable/twin/coercion'
```

And finally, following my example, my form class now looks something like this:

```ruby
    class ContactForm < ::Reform::Form
      include ContactModule
      include Disposable::Twin::Coercion
      feature Coercion
    end
```

**Comment by fran-worley** [comment]
@adrienbourgeois. You should be able to use the coercion module from reform as shorthand:

https://github.com/apotonick/reform/blob/master/lib/reform/form/coercion.rb

You only need to use feature instead of include if you need that module to also be included in collections or nested properties of your form otherwise you can just use include.

You could add your default modules to reform via an initialiser examples of how to do this are on the trailblazer reform page. 

If you've got more questions come find us in the gitter room!
